pydantic/_internal/_known_annotated_metadata.py

```diff
@@ -239,7 +239,25 @@ def apply_known_metadata(annotation: Any, schema: CoreSchema) -> CoreSchema | No
 
         #  else, apply a function after validator to the schema to enforce the corresponding constraint
         if constraint in chain_schema_constraints:
-            chain_schema_steps.append(cs.str_schema(**{constraint: value}))
+
+            def _apply_constraint_with_incompatibility_info(
+                value: Any, handler: cs.ValidatorFunctionWrapHandler
+            ) -> Any:
+                try:
+                    x = handler(value)
+                except ValidationError as ve:
+                    if 'type' in ve.errors()[0]['type']:
+                        raise TypeError(
+                            f"Unable to apply constraint '{constraint}' to supplied value {value} for schema of type '{schema_type}'"  # noqa: B023
+                        )
+                    raise ve
+                return x
```

Comment:
Maybe add an example here in a comment? It's not clear to me what the goal is just looking at hte code

Reply:
Done, thanks for the tip!